src/content/containers/Home.js

```diff
@@ -33,6 +33,27 @@ type InstallButtonProps = {
   className?: string,
 };
 
+type UploadButtonProps = {
+  retrieveProfileFromFile: File => void,
+};
+
+class UploadButton extends PureComponent {
+  props: UploadButtonProps;
+
+  render() {
+    return (
+      <div>
+        <input type="file" ref="file"></input>
+        <button onClick={e => this._upload()}>Upload</button>
```

Comment:
I think the separate button is unnecessary. I'd simply call `_upload` on the input's `onChange` event. What do you think ?

This would have the nice side effect of preventing errors like the user pressing the button without selecting a file.